### PR TITLE
Avoid unnecessary char casts for CookieEncoder

### PR DESCRIPTION
**Pull request by doom369**
Motivation:

Avoid unnecessary (char) casts by changing variables types.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
     /**
      * Equals '='
      */
-    public static final byte EQUALS = 61;
+    public static final char EQUALS = '=';
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: thats a breaking change so it cant be done in 4.1

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java [hunk 2]
```diff
@@ -64,7 +64,7 @@
     /**
      * Double quote '"'
      */
-    public static final byte DOUBLE_QUOTE = '"';
+    public static final char DOUBLE_QUOTE = '"';
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: same as above

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java [hunk 3]
```diff
@@ -54,7 +54,7 @@
     /**
      * Semicolon ';'
      */
-    public static final byte SEMICOLON = 59;
+    public static final char SEMICOLON = ';';
```
Comment (normanmaurer) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: same as above

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java (hunk 1 above)
Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: Yeah, thought about that after submission. 5.0?

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java (hunk 2 above)
Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: @normanmaurer fixed.

**Comment by normanmaurer** [comment]
@netty-bot test this please

File: codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java [hunk 4]
```diff
@@ -97,24 +97,24 @@ static String stripTrailingSeparator(StringBuilder buf) {
 
     static void add(StringBuilder sb, String name, long val) {
         sb.append(name);
-        sb.append((char) HttpConstants.EQUALS);
+        sb.append(HttpConstants.EQUALS_CHAR);
```
[the remark this answers is not in the document]
Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java]: I agree, just followed the used style.

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java [hunk 5]
```diff
@@ -41,6 +41,8 @@
      */
     public static final byte EQUALS = 61;
 
+    public static final char EQUALS_CHAR = '=';
```
[the remark this answers is not in the document]
Comment (normanmaurer) [reply codec-http/src/main/java/io/netty/handler/codec/http/HttpConstants.java]: Thats a good point @ejona86

File: codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java]: @normanmaurer WDYT? Should I replace with char literals? Seems reasonable to me.

[the remark this answers is not in the document]
Comment (normanmaurer) [reply codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java]: sounds good.

[the remark this answers is not in the document]
Comment (doom369) [reply codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieUtil.java]: done

**Comment by normanmaurer** [comment]
@doom369 thanks!

**Pull request merged**